packages/relic/lib/src/headers/special/accept_encoding_header.dart

```diff
@@ -0,0 +1,55 @@
+part of '../headers.dart';
+
+/// A class representing the HTTP Accept-Encoding header.
+///
+/// This header specifies the content encoding that the client can understand.
+class AcceptEncodingHeader {
+  /// The list of encodings that are accepted.
+  final List<EncodingQuality> encodings;
+
+  /// Constructs an instance of [AcceptEncodingHeader] with the given encodings.
+  AcceptEncodingHeader(this.encodings);
+
+  /// Parses the Accept-Encoding header value and returns an [AcceptEncodingHeader] instance.
+  factory AcceptEncodingHeader.parse(List<String> values) {
+    var parts = values.splitTrimAndFilterUnique();
+
+    if (parts.isEmpty) {
+      throw FormatException(
+        'Invalid Accept-Encoding header: value cannot be empty',
+      );
+    }
+
+    var encodings = parts.map((e) {
+      var parts = e.split(';q=');
```

Comment:
Suggestion(nit): Defininig a variable named parts while iterating a map called parts. Please clarify what is what.